subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/transform/DefaultArtifactTransformsTest.groovy

```diff
@@ -50,7 +50,18 @@ class DefaultArtifactTransformsTest extends Specification {
     def consumerSchema = Mock(AttributesSchemaInternal)
     def attributeMatcher = Mock(AttributeMatcher)
     def dependenciesResolver = Stub(ExtraExecutionGraphDependenciesResolverFactory)
-    def transforms = new DefaultArtifactTransforms(matchingCache, consumerSchema, AttributeTestUtil.attributesFactory())
+    def transformationNodeFactory = new TransformationNodeFactory() {
+        @Override
+        Collection<TransformationNode> getOrCreate(ResolvedArtifactSet artifactSet, Transformation transformation, ExecutionGraphDependenciesResolver dependenciesResolver) {
+            throw new UnsupportedOperationException()
+        }
+
+        @Override
+        Try<TransformationSubject> getResultIfCompleted(ComponentArtifactIdentifier artifactId, Transformation transformation) {
+            return null
+        }
+    }
```

Comment:
I think a mock would be much nicer here. It would also signal if we accidentally called this thing despite the expectation that the test doesn't use it.

Reply:
Using a mock now.